lighthouse-core/lib/proto-preprocessor.js

```diff
@@ -16,11 +16,12 @@ const fs = require('fs');
  */
 
 /**
-  * @param {string} result
+  * Transform an LHR into a proto-friendly LHR
+  * @param {LH.Result} lhr
+  * @return {LH.Result}
   */
-function processForProto(result) {
-  /** @type {LH.Result} */
-  const reportJson = JSON.parse(result);
+function processForProto(lhr) {
+  const reportJson = JSON.parse(JSON.stringify(lhr));
```

Comment:
could we skip the cloning? Only the lr entry uses this fn, and the original lhr is not needed there. if some other client comes along with different requirements they can clone the input themselves.

assuming cloning a lhr takes a measurable amount of time.

I clocked 4ms for example.com and for 8ms nyt.com:

```
console.time('clone'); JSON.parse(JSON.stringify(lhr)); console.timeEnd('clone');
```

Reply:
I find it unexpected when methods mutate the passed in object, so this is my preferred pattern. It's also what we do for Util.prepareReportResult, swapLocale, our tests, etc.

We adopted this pattern in our tests, as previously we had some unexpected mutations that caused failures that were tricky to track down.

Reply:
yeah I prefer avoiding mutation too. if the few ms is worth the change, I'd suggest renaming the method to `modifyForProto` to properly set expectations.